src/framework/ui/internal/platform/windows/windowsplatformtheme.cpp

```diff
@@ -0,0 +1,139 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2021 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+
+#include "windowsplatformtheme.h"
+#include "log.h"
+#include <Windows.h>
+
+using namespace mu::ui;
+using namespace mu::async;
+
+using ThemeType = IUiConfiguration::ThemeType;
+
+static const std::wstring windowsThemeKey = L"Software\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize";
+static const std::wstring windowsThemeSubkey = L"AppsUseLightTheme";
+
+HKEY hKey = nullptr;
+
+WindowsPlatformTheme::WindowsPlatformTheme()
+{
+    using fnRtlGetNtVersionNumbers = void(WINAPI*)(LPDWORD major, LPDWORD minor, LPDWORD build);
+    auto rtlGetNtVersionNumbers
+        = reinterpret_cast<fnRtlGetNtVersionNumbers>(GetProcAddress(GetModuleHandleW(L"ntdll.dll"), "RtlGetNtVersionNumbers"));
+    if (rtlGetNtVersionNumbers) {
+        DWORD major, minor, buildNumber;
+        rtlGetNtVersionNumbers(&major, &minor, &buildNumber);
+        m_buildNumber = buildNumber & ~0xF0000000;
+    }
+}
+
+WindowsPlatformTheme::~WindowsPlatformTheme()
+{
+    stopListening();
+}
+
+void WindowsPlatformTheme::init()
+{
+    updateListeningStatus(configuration()->preferredThemeType());
+
+    configuration()->preferredThemeTypeChanged().onReceive(nullptr, [this](const ThemeType& newThemeType) {
+        updateListeningStatus(newThemeType);
+    });
+}
+
+void WindowsPlatformTheme::updateListeningStatus(ThemeType themeType)
+{
+    if (themeType == ThemeType::FOLLOW_SYSTEM_THEME) {
+        startListening();
+    } else {
+        stopListening();
+    }
+}
+
+void WindowsPlatformTheme::startListening()
+{
+    if (isListening) {
+        return;
+    }
+    isListening = true;
+    if (RegOpenKeyEx(HKEY_CURRENT_USER, windowsThemeKey.c_str(), 0,
+                     KEY_NOTIFY | KEY_CREATE_SUB_KEY | KEY_ENUMERATE_SUB_KEYS | KEY_QUERY_VALUE | KEY_WOW64_64KEY,
+                     &hKey) == ERROR_SUCCESS) {
+        std::thread listenThread([this]() { listen(); });
+        listenThread.detach();
+    } else {
+        LOGD() << "Failed opening key for listening dark theme changes.";
+    }
+}
+
+void WindowsPlatformTheme::listen()
```

Comment:
better rename to `th_listen` (this is not rule, but this helpful for easy understand)